landscape/client/manager/tests/test_scriptexecution.py

```diff
@@ -426,6 +437,10 @@ def _run_script(self, username, uid, gid, path):
         patch_chown = mock.patch("os.chown")
         mock_chown = patch_chown.start()
 
+        patch_issnap = mock.patch("landscape.client.IS_SNAP")
+        patch_issnap.return_value = from_snap
+        patch_issnap.start()
```

Comment:
Not sure _exactly_ why this is working, but `return_value` shoudl be set on the return value of `start`, not patch:
```python
patch_issnap = mock.patch("landscape.client.IS_SNAP")
mock_issnap = patch_issnap.start()
mock_issnap.return_value = from_snap
```
I guess the return value isn't being used/checked in any meaningful way in the test?

Reply:
The return value is just being used to ensure we hit both control paths of the code which we can see is working from the coverage checks. I'll have a go at changing it over and have a rummage through the documentation on patching a global variable and see if there is anything noteworthy.

Reply:
So, you are right, it wasn't actually working. The patch was not being applied appropriately to an imported global variable - which then overshadowed a different bug in the test. 

To fix that I've tweaked the scriptexecution class so that the import of IS_SNAP is then copied to a local (to the object) variable that the test script can then actually change without needing to patch anything.

Reply:
Also, as IS_SNAP will default to None, rather than False, the tests should reflect that.

Reply:
To shed some light on the "patch was not being applied appropriately": `patch` hooks itself into variable _lookup_, not definition, so what you really want to patch is the name of the variable where it's used. In this case, that'd be
```python
mock.patch("landscape.client.manager.scriptexecution.IS_SNAP")
```
totally forgiven for not knowing this little bit of Python errata, and I'm okay with how you're doing it now.